src/geode/model/representation/core/detail/transfer_collections.cpp

```diff
@@ -0,0 +1,378 @@
+        const geode::BRep& new_model,
+        typename geode::BRepBuilder& new_model_builder,
+        const geode::ModelGenericMapping& component_mapping )
+    {
+        if( !component_mapping.has_mapping_type(
+                geode::Block3D::component_type_static() ) )
+        {
+            return;
+        }
+        const auto& blocks_mapping =
+            component_mapping.at( geode::Block3D::component_type_static() );
+        for( const auto& collection_in : old_model.block_collections() )
+        {
+            bool exists_in_out{ false };
+            for( const auto& block_in_collection :
+                old_model.block_collection_items( collection_in ) )
+            {
+                if( blocks_mapping.has_mapping_input(
+                        block_in_collection.id() ) )
+                {
+                    exists_in_out = true;
+                    break;
+                }
+            }
+            if( !exists_in_out )
+            {
+                continue;
+            }
+            new_model_builder.add_block_collection( collection_in.id() );
+            new_model_builder.set_block_collection_name(
+                collection_in.id(), collection_in.name() );
+            const auto& new_collection =
+                new_model.block_collection( collection_in.id() );
+            for( const auto& block_in_collection :
+                old_model.block_collection_items( collection_in ) )
+            {
+                if( !blocks_mapping.has_mapping_input(
+                        block_in_collection.id() ) )
+                {
+                    continue;
+                }
+                for( const auto& block_out_id :
+                    blocks_mapping.in2out( block_in_collection.id() ) )
+                {
+                    new_model_builder.add_block_in_block_collection(
+                        new_model.block( block_out_id ), new_collection );
+                }
+            }
+        }
+    }
+} // namespace
+
+namespace geode
+{
+    namespace detail
+    {
+        void opengeode_model_api transfer_brep_collections(
+            const BRep& old_brep,
+            const BRep& new_brep,
+            BRepBuilder& new_brep_builder,
+            const ModelGenericMapping& component_mapping )
+        {
+            transfer_corner_collections(
+                old_brep, new_brep, new_brep_builder, component_mapping );
+            transfer_line_collections(
+                old_brep, new_brep, new_brep_builder, component_mapping );
+            transfer_surface_collections(
+                old_brep, new_brep, new_brep_builder, component_mapping );
+            transfer_block_collections(
+                old_brep, new_brep, new_brep_builder, component_mapping );
+            if( !component_mapping.has_mapping_type(
+                    geode::Surface3D::component_type_static() ) )
+            {
+                return;
+            }
+            const auto& surfaces_mapping =
```

Comment:
add also a function for this, even if it is used once, so the transfer_brep_collections() has only one level of abstraction